app/src/main/java/io/edanni/movies/domain/service/MovieService.kt

```diff
@@ -0,0 +1,117 @@
+package io.edanni.movies.domain.service
+
+import io.edanni.movies.infrastructure.api.MovieApi
+import io.edanni.movies.infrastructure.api.dto.Configuration
+import io.edanni.movies.infrastructure.api.dto.Genre
+import io.edanni.movies.infrastructure.api.dto.Movie
+import io.edanni.movies.infrastructure.api.dto.MoviePage
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.rxkotlin.Observables
+import retrofit2.Retrofit
+import javax.inject.Inject
+
+/**
+ *
+ */
+class MovieService
+@Inject constructor(retrofit: Retrofit) {
+    private val movieApi: MovieApi = retrofit.create(MovieApi::class.java)
```

Comment:
This was an oversight. In hindsight, having it injected by Dagger would ease testing, since I could simply mock the response instead of mocking the entire HTTP request.